app/models/package_manager/base.rb

```diff
@@ -177,16 +177,8 @@ def self.deprecate_versions(db_project, version_hashes)
           .where.not(number: version_hashes.pluck(:number))
           .where("status != ? or status is null", removed_status)
           .update_all(status: removed_status, updated_at: current_time)
-      when "pypi"
-        removed_versions = version_hashes.filter { |x| x[:yanked] == true }
-
-        if removed_versions.any?
-          db_project
-            .versions
-            .where(number: removed_versions.pluck(:number).compact)
-            .where("status != ? or status is null", removed_status)
-            .update_all(status: removed_status, updated_at: current_time)
-        end
+        # yanked pypi versions are marked as such in the api, and the majority of them are handled upstream of here
```

Comment:
This change basically takes advantage of the fact that the `add_version` call on line 133 does both an add (for versions we haven't seen before) or an update (for versions that we know about already). The latter means all the data that gets passed in needs to be current state of the version row, so the status change can be passed in here.